swap_meet/vendor.py

```diff
@@ -1,2 +1,101 @@
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+        else:
+            return None
+        return item
+
+    def get_by_id(self, id):      
+        for item in self.inventory:
+            if id == item.id:
+                return item
+        return None
+
+    def swap_items(self, other_vendor, my_item, their_item):       
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+
+        other_vendor.add(self.remove(my_item))
+        self.add(other_vendor.remove(their_item))
+
+        return True
+        
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        result = self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
```

Comment:
Thank you for suggesting a swap in place to obtain O(1) time but I don't understand this. 
`self.inventory[0], other_vendor.inventory[0] = other_vendor.inventory[0], self.inventory[0]`
I used `temp` variable to hold the value of `self.inventory[0]` and swap it with `other_vendor.inventory[0]`
`temp = self.inventory[0]
 self.inventory[0] = other_vendor.inventory[0]
 other_vendor.inventory[0] = temp`

Reply:
What we're doing is called "multiple assignment" or sometimes is referred to as "tuple unpacking", because it commonly is used with tuples. I suggest [taking a look at this article](https://treyhunner.com/2018/03/tuple-unpacking-improves-python-code-readability/#:~:text=Multiple%20assignment%20(also%20known%20as,when%20you%20need%20it%20most.) for some more examples. The general idea is that you can list multiple values on the left, and new values for those variables on the right. The right side of the `=` operator is evaluated first, so those values are held in memory and then those values get applied to the variables on the left.

Reply:
Thank you for the explanation! But when I tried the code, it failed test_integration_wave_04_05_06 and skipped  test_integration_wave_01_02_03

Reply:
The tests should fail in this particular scenario since the integration tests that exercise the choose_and_swap_items method is expecting the inventory to be in a particular order, I just wanted to highlight that we can swap items in this way.